source/common/common/utility.h

```diff
@@ -110,6 +111,22 @@ class RealTimeSource : public TimeSource {
   MonotonicTime monotonicTime() override { return std::chrono::steady_clock::now(); }
 };
 
+/**
+ * Class used for creating non-memory allocating std::ostream.
+ */
+class FixedSizeStreamBuffer : public std::streambuf {
+public:
+  FixedSizeStreamBuffer(char* base, size_t size);
+};
+
+/**
+ * std::ostream class that serializes writes into the provided buffer.
+ */
+class OutputBufferStream : public virtual FixedSizeStreamBuffer, public std::ostream {
```

Comment:
This class should probably have a way to provide the number of bytes that were written into the buffer.

Reply:
Done.